### PR TITLE
[WebRTC] Add new DataChannel buffered amount APIs

### PR DESCRIPTION
**Pull request by hsgwon**
### Description of Change ###

Add new DataChannel buffered amount APIs

### API Changes ###

 - TCSACR-480

### CODE REVIEW

File: src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs [hunk 1]
```diff
@@ -113,6 +113,39 @@ private IntPtr Handle
...
         public string Label { get; }
 
+        private uint? _bufferThreshold;
+        /// <summary>
+        /// Gets or sets the threshold of data channel buffered amount.<br/>
+        /// If the amount of buffered data is lower than threshold value, <see cref="BufferUnderflow"/> will be occurred.<br/>
+        /// The default value is 0. and if threshold is 0, <see cref="BufferUnderflow"/> is not occurred.
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        public uint BufferThreshold
```
[the remark this answers is not in the document]
Comment (sangchul1011) [reply src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs:123]: typo: BufferedAmountThreshold -> BufferedAmountLowThreshold

File: src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs [hunk 2]
```diff
@@ -113,6 +113,39 @@ private IntPtr Handle
...
+            {
+                ValidateNotDisposed();
+
+                _bufferThreshold = value;
+
+                SetDataChannelBufferedAmountLowThresholdCallback();
+            }
+        }
+
         /// <summary>
```
Comment (sangchul1011) [inline src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs:149]: Also we need to add prop of BufferedAmount.

File: src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.Events.cs [hunk 3]
```diff
@@ -132,6 +134,31 @@ public partial class WebRTCDataChannel
             }
         }
 
+        /// <summary>
+        /// Occurs when the buffered amount is lower than <see cref="BufferThreshold"/>.<br/>
+        /// If <see cref="BufferThreshold"/> is not set, this event will not occurred.
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        public event EventHandler<EventArgs> BufferUnderflow
```
[the remark this answers is not in the document]
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.Events.cs:142]: ok

File: src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs]: It's already changed.

File: src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs (hunk 2 above)
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs:167]: It's already added.

File: src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs [hunk 4]
```diff
@@ -113,14 +113,32 @@ private IntPtr Handle
...
         /// Gets or sets the threshold of data channel buffered amount.<br/>
         /// If the amount of buffered data is lower than threshold value, <see cref="BufferUnderflow"/> will be occurred.<br/>
         /// The default value is 0. and if threshold is 0, <see cref="BufferUnderflow"/> is not occurred.
         /// </summary>
         /// <since_tizen> 10 </since_tizen>
-        public uint BufferThreshold
+        public uint BufferedAmountLowThreshold
         {
             get
             {
```
Comment (sangchul1011) [inline src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs:144]: log of line 150 might be
 'failed to buffered amount threshold value'.

Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs:144]: ok i'll fix it.

File: src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs [hunk 5]
```diff
@@ -113,6 +113,57 @@ private IntPtr Handle
...
+
+                return amount;
+            }
+        }
+
+        private uint? _bufferThreshold;
+        /// <summary>
+        /// Gets or sets the threshold of data channel buffered amount.<br/>
+        /// If the amount of buffered data is lower than threshold value, <see cref="BufferedAmountLow"/> will be occurred.<br/>
+        /// The default value is 0. and if threshold is 0, <see cref="BufferedAmountLow"/> is not occurred.
```
Comment (wiertel) [inline src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs:138]: after dot "." new sentence should be started with capital letter,  "be occurred" should replaced by "be raised" or sth like that.
```suggestion
        /// The default value is 0, which means <see cref="BufferedAmountLow"/> is disabled and will not be raised.
```

File: src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.Events.cs [hunk 6]
```diff
@@ -132,6 +134,31 @@ public partial class WebRTCDataChannel
             }
         }
 
+        /// <summary>
+        /// Occurs when the buffered amount is lower than <see cref="BufferedAmountLowThreshold"/>.<br/>
+        /// If <see cref="BufferedAmountLowThreshold"/> is not set, this event will not occurred.
```
Comment (wiertel) [inline src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.Events.cs:139]: "will not occurred"  -> will not be raised / will not be triggered / will not occur 
```suggestion
        /// Occurs when the buffered data amount is lower than <see cref="BufferedAmountLowThreshold"/>.<br/>
        /// If <see cref="BufferedAmountLowThreshold"/> is not set, this event will not be raised.
```

File: src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.Events.cs [hunk 7]
```diff
@@ -132,6 +134,31 @@ public partial class WebRTCDataChannel
...
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        public event EventHandler<EventArgs> BufferedAmountLow
```
Comment (wiertel) [inline src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.Events.cs:142]: I'm not convinced about the event name, nothing in that name suggests it's an event. Unfortunately, the names I can think of like `BufferedAmountDroppedBelowThreshold` or `BufferedAmountFallenBelowTheshold` or `BufferDrained` are really better, so I think I will accept the current name as it is.

File: src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs [hunk 8]
```diff
@@ -113,6 +113,57 @@ private IntPtr Handle
...
+                    ThrowIfFailed("Failed to get buffered amount");
+
+                return amount;
+            }
+        }
+
+        private uint? _bufferThreshold;
+        /// <summary>
+        /// Gets or sets the threshold of data channel buffered amount.<br/>
+        /// If the amount of buffered data is lower than threshold value, <see cref="BufferedAmountLow"/> will be occurred.<br/>
```
[the remark this answers is not in the document]
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs:137]: Okay. I'll fix it.

File: src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs (hunk 5 above)
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.cs:138]: Okay. I'll fix it.

File: src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.Events.cs (hunk 6 above)
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.Events.cs:139]: Okay. I'll fix it.

File: src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.Events.cs (hunk 7 above)
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.Events.cs:142]: The word `BufferedAmountLow` is used in the WebRTC spec. So we want to follow it.

Comment (wiertel) [reply src/Tizen.Multimedia.Remoting/WebRTC/WebRTCDataChannel.Events.cs:142]: ok, let's keep it as it is

**Pull request closed without merging**